File: bubblesort.py
arr = [6, 54, 5, 125, 222, 113, 0]

def bubbleSort(arr):
    n = len(arr)
    for x in range(n-1):
        for i in range(0, n-x-1):
            if arr[i] > arr[i + 1] :
                arr[i], arr[i + 1] = arr[i + 1], arr[i]


bubbleSort(arr)
print ("Sorted array ")
for x in range(len(arr)):
    print ("% d" % arr[x]),

File: bubblesort.cpp
#include <bits/stdc++.h>
using namespace std;
 
void swap(int *num1, int *num2)
{
    int temp = *num1;
    *num1 = *num2;
    *num2 = temp;
}
 
void bubbleSort(int arr[], int n)
{
    int i, j;
    for (i = 0; i < n-1; i++)    
     
    
    for (j = 0; j < n-i-1; j++)
        if (arr[j] > arr[j+1])
            swap(&arr[j], &arr[j+1]);
}
 

void printarray(int arr[], int size)
{
    int i;
    for (i = 0; i < size; i++)
        cout << arr[i] << " ";
    cout << endl;
}
 
int main()
{
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr)/sizeof(arr[0]);
    bubbleSort(arr, n);
    cout<<"Sorted array--";
    printarray(arr, n);
    return 0;
}
 